docs/references/annotations.md

```diff
@@ -309,6 +345,22 @@ sets the
 [`service.client_certificate`](https://docs.konghq.com/latest/admin-api/#service-object)
 for the service.
 
+### `konghq.com/host-header`
+
+> Available since controller 0.9
+
+Sets the `host_header` setting on the Kong upstream created to represent a
+Kubernetes Service. By default, Kong upstreams set `Host` to the hostname or IP
+address of an individual target (the Pod IP for controller-managed
+configuration). This annotation overrides the default behavior and sends
```

Comment:
I'm not sure if this is correct. `PreserveHost` is set to `true` by default, meaning the `host` header will be what Kong receives in the downstream request.

Reply:
Kind of in a weird open question: these settings do interact, but it's not clear how they're supposed to. https://github.com/Kong/kong/issues/5580 is still open and has gone a bit stale, but indicates that your PoV is that this setting should take precedence. Do you think there will be any movement there, or will that likely remain as-is (`host_header` is only used if `preserve_host` is off)?

Reply:
>  Do you think there will be any movement there, or will that likely remain as-is (host_header is only used if preserve_host is off)?

Not sure when that will start moving that's why I recommend document what stands true today.